src/ode_solver/ode_solver_factory.cpp

```diff
@@ -1,39 +1,26 @@
 #include "ode_solver_factory.h"
 #include "parameters/all_parameters.h"
 #include "ode_solver_base.h"
-#include "explicit_ode_solver.h"
+#include "runge_kutta_ode_solver.h"
 #include "implicit_ode_solver.h"
 #include "rrk_explicit_ode_solver.h"
 #include "pod_galerkin_ode_solver.h"
 #include "pod_petrov_galerkin_ode_solver.h"
 #include <deal.II/distributed/solution_transfer.h>
+#include "runge_kutta_methods/runge_kutta_methods.h"
+#include "runge_kutta_methods/rk_tableau_base.h"
 
 namespace PHiLiP {
 namespace ODE {
 
 template <int dim, typename real, typename MeshType>
 std::shared_ptr<ODESolverBase<dim,real,MeshType>> ODESolverFactory<dim,real,MeshType>::create_ODESolver(std::shared_ptr< DGBase<dim,real,MeshType> > dg_input)
 {
-    std::cout << "Creating ODE Solver..." << std::endl;
     using ODEEnum = Parameters::ODESolverParam::ODESolverEnum;
     ODEEnum ode_solver_type = dg_input->all_parameters->ode_solver_param.ode_solver_type;
-    if(ode_solver_type == ODEEnum::explicit_solver)        return std::make_shared<ExplicitODESolver<dim,real,MeshType>>(dg_input);
-    if(ode_solver_type == ODEEnum::implicit_solver)        return std::make_shared<ImplicitODESolver<dim,real,MeshType>>(dg_input);
-    if constexpr(dim==1){
-        //RRK is only implemented for Burgers on collocated nodes, 1D
-        const bool use_collocated_nodes = dg_input->all_parameters->use_collocated_nodes;
-        using PDEEnum = Parameters::AllParameters::PartialDifferentialEquation;
-        const PDEEnum pde_type = dg_input->all_parameters->pde_type;
-        const bool use_inviscid_burgers = (pde_type == PDEEnum::burgers_inviscid);
-        if ((ode_solver_type == ODEEnum::rrk_explicit_solver) && 
-                use_collocated_nodes && use_inviscid_burgers){
-            return std::make_shared<RRKExplicitODESolver<dim,real,MeshType>>(dg_input);
-        }
-        else{
-            display_error_ode_solver_factory(ode_solver_type, false);
-            return nullptr;
-        }
-    }
+    if(ode_solver_type == ODEEnum::runge_kutta_solver)         return create_RungeKuttaODESolver(dg_input);
+    if(ode_solver_type == ODEEnum::rrk_explicit_solver)        return create_RungeKuttaODESolver(dg_input);
```

Comment:
maybe combine these if statements so were only calling that RK factory once

Reply:
could introduce a bool `is_runge_kutta_type_solver`

Reply:
I have combined them in the most recent commit. I think that it is okay to have two conditions rather than a bool `is_runge_kutta_type_solver`, but I definitely will add a bool if I have to add any more RK-based ODE solvers in the future.